File: backend/src/main/java/net/wuxianjie/backend/shared/operationlog/OperationLog.java
package net.wuxianjie.backend.shared.operationlog;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志数据库表。
 */
@Data
public class OperationLog {

  /**
   * 日志 ID。
   */
  private Long id;

  /**
   * 请求时间。
   */
  private LocalDateTime requestedAt;

  /**
   * 客户端 IP。
   */
  private String clientIp;

  /**
   * 用户名。
   */
  private String username;

  /**
   * 操作描述。
   */
  private String message;
}

File: frontend/src/shared/apis/backend/operation-log.ts
import { requestApi } from '@/shared/apis/backend/helpers'
import type { PaginationData, PaginationParams } from '@/shared/apis/types'

export type GetLogsParams = PaginationParams & {
  startAt: string
  endAt: string
  clientIp?: string
  username?: string
  message?: string
}

export type Log = {
  id: number
  requestedAt: string
  clientIp: string
  username: string
  message: string
}

type ChartDataItem = {
  name: string
  value: number
}

export type ChartData = ChartDataItem[]

/**
 * 获取操作日志分页数据。
 *
 * @param params 分页查询条件
 * @returns Promise 响应结果
 */
export async function getLogsApi(params: GetLogsParams) {
  return await requestApi<PaginationData<Log>>({
    url: '/api/v1/operation-logs',
    urlParams: params
  })
}

/**
 * 获取登录数前几的用户。
 *
 * @param num 前几
 * @returns Promise 响应结果
 */
export async function getLoginsTopApi(num: number) {
  return await requestApi<ChartData>({
    url: `/api/v1/operation-logs/logins-top/${num}`
  })
}

/**
 * 获取最近几天的登录数。
 *
 * @param days 最近几天
 * @returns Promise 响应结果
 */
export async function getLoginsHistoryApi(days: number) {
  return await requestApi<ChartData>({
    url: `/api/v1/operation-logs/logins-history/${days}`
  })
}
